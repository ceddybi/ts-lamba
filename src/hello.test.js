import helloError from './hello-ts-error';
import helloSuccess from './hello-success';
import newrelic from 'newrelic';

describe('NewRelic Lambda', () => {

  describe('given a lambda that executes successfully', () => {
    let spyStartBackgroundTransaction;
    beforeEach(async () => {
      spyStartBackgroundTransaction = jest.spyOn(newrelic, 'startBackgroundTransaction');
    });

    afterEach(async () => {
      spyStartBackgroundTransaction.mockRestore();
    });

    it('should run successfully with response', async () => {
      const response = await helloSuccess({});
      expect(response).toMatchSnapshot();
    });

    it('should call "startBackgroundTransaction" ', async () => {
      await helloSuccess({});
      expect(spyStartBackgroundTransaction).toHaveBeenCalled();
    });

  });

  describe('given a lambda that executes with an exception', () => {

    let spyNoticeError;

    beforeEach(async () => {
      spyNoticeError = jest.spyOn(newrelic, 'noticeError');
    });

    afterEach(async () => {
      spyNoticeError.mockRestore();
    });

    it('should execute and throw an exception', async () => {
      let response;
      let err;

      try {
        response = await helloError();
      }
      catch (error) {
        err = error;
      }

      expect(err.message).toEqual("Error with unsent");

    });

    it('should call "noticeError" when an exception occurs', async () => {
      let response;
      let err;

      try {
        response = await helloError();
      }
      catch (error) {
        err = error;
      }

      expect(spyNoticeError).toHaveBeenCalled();

    });

  });

});
