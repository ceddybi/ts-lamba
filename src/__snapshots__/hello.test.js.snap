// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRelic Lambda given a lambda that executes successfully should run successfully with response 1`] = `null`;

exports[`hello executes as expected 1`] = `
Object {
  "body": "{\\"message\\":\\"Go Serverless! Your function executed successfully!\\",\\"input\\":{}}",
  "statusCode": 200,
}
`;
